zerver/lib/export.py

```diff
@@ -1425,7 +1425,13 @@ def do_import_realm(import_dir):
 
     convert_to_id_fields(data, 'zerver_realm', 'notifications_stream')
     fix_datetime_fields(data, 'zerver_realm')
-    realm = Realm(**data['zerver_realm'][0])
+    zr = data['zerver_realm'][0]
+    # Convert the authentication methods from list of (name, bool) to bitfield
+    auth_methods_in_bitfield = ''.join(['1' if a[1] else '0' for a in zr["authentication_methods"]])
+    # Convert the authentication methods to integer
+    auth_method_in_int = int(auth_methods_in_bitfield, 2)
+    zr["authentication_methods"] = auth_method_in_int
+    realm = Realm(**zr)
```

Comment:
Do we also need to change the `export` code path to export them in the right format?

Reply:
Also, it might be more robust to iterate through `AUTHENTICATION_FLAGS` rather than assuming the export has the fields in order?

Can you please move this fixup code into a function?  That's the style we use in this module.

Reply:
No, the `export` code path is already human-readable, in the form of `[["Google", true], ["Email", true], ["Github", true], ["LDAP", true], ["Dev", true], ["RemoteUser", true]]`. The purpose of this PR is to keep the auth_methods part of the export remain human-readable.